rclcpp/test/rclcpp/node_interfaces/test_node_graph.cpp

```diff
@@ -95,6 +95,9 @@ TEST_F(TestNodeGraph, get_service_names_and_types_by_node)
   EXPECT_THROW(
     node_graph->get_service_names_and_types_by_node("not_a_node", "not_absolute_namespace"),
     std::runtime_error);
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
The main reason for this test was just to add coverage for the bad arguments above, but I still wanted to add a nominal case. Is there perhaps a better check I could be doing here instead of just checking that the two are the same and can avoid waiting for a full second to establish the node graph discovery?

Reply:
The test needs to wait until node 1 has discovered node 2. It could do so iteratively to finish as early as possible with a maximum timeout. Please feel free to commit directly to this branch to update the test.

Reply:
Will do